### ISSUE TITLE
How to combine loading of systemjs and extras in page

### ISSUE THREAD

**Issue by kouts** [systemjs/systemjs#1926]
Great work on systemjs, I'm using the latest version (3.1) and I'm including the following
files in my webpage via multiple script tags.
s.js
system.js
extras/amd.js
extras/global.js
extras/named-exports.js
extras/named-register.js
extras/transform.js

All of these files are needed for my application to run.
Now I'm looking for a away to combine all these files into one let's say "bootstrap.js" file,
which might also contain some polyfills.
Right now the only way to do it is manually concatenate these files.
I'm using Webpack and I tried importing the files one by one e.g
```
import 'systemjs/dist/s.js'
import 'systemjs/dist/system.js'
import 'systemjs/dist/extras/amd.js
...
```
But this doesn't work, I'm getting error systemRegister is undefined.
Is there a way to load systemjs via ```import``` or am I missing something obvious?

**Comment by guybedford** [comment]
I think you just want the following included:

```
system.js
extras/amd.js
extras/named-exports.js
extras/named-register.js
extras/transform.js
```

Because system.js already include s.js and extras/global.js

It should work with this modular build. What is the specific error that you are seeing here?

When building with Webpack did you make sure to add the parser rule for the System global (https://github.com/systemjs/systemjs#compatibility-with-webpack)?

**Comment by guybedford** [comment]
I think you just want the following included:

```
system.js
extras/amd.js
extras/named-exports.js
extras/named-register.js
extras/transform.js
```

Because system.js already include s.js and extras/global.js

It should work with this modular build. What is the specific error that you are seeing here?

When building with Webpack did you make sure to add the parser rule for the System global (https://github.com/systemjs/systemjs#compatibility-with-webpack)?

**Comment by kouts** [comment]
@guybedford thanks for the tip.
What I'm getting is ```Uncaught TypeError: Cannot read property 'length' of undefined```
![error](https://user-images.githubusercontent.com/2233663/56083370-a87f5580-5e2c-11e9-95db-5788bbc715b4.png)
I'm importing systemjs modules with webpack like this:
```
import 'systemjs/dist/system.js';
import 'systemjs/dist/extras/amd.js';
import 'systemjs/dist/extras/named-exports.js';
import 'systemjs/dist/extras/named-register.js';
import 'systemjs/dist/extras/transform.js';
```

**Comment by kouts** [comment]
I have created a [test repo here](https://github.com/kouts/systemjs-webpack) showcasing the issue.
The src/bootstrap.js file is importing ```systemjs``` with standard ES6 import.
You can see the error on the demo page.

**Comment by guybedford** [comment]
@kouts did you try setting the following webpack rule here?

```js
{
  module: {
    rules: [
      { parser: { system: false } }
    ]
  }
}
```

**Comment by kouts** [comment]
yes I have added this to the [webpack configuration](https://github.com/kouts/systemjs-webpack/blob/master/webpack.common.js)

**Comment by guybedford** [comment]
Ok I had a look at the bundle, and the problem is that Webpack still seems to be injecting the `System` scoped reference to its own version of System here - https://github.com/kouts/systemjs-webpack/blob/master/dist/js/bootstrap.bundle.js#L92.

This is then interfering with the AMD extra which expects to be able to attach to just `System` which is now actually the Webpack loader.

The work-around is to edit the AMD extra to contain the following second line:

```js
 const systemPrototype = global.System.constructor.prototype;
```

But you will get problems as soon as you reference the `System` global yourself - `System.import` will still reference the webpack loader.

This sounds to me like the parser rule above isn't applying properly. I'm not sure why that would be, but would be the root cause to work out to ensure that module 0 isn't emitted by Webpack.

**Comment by kouts** [comment]
So is this a bug with Webpack, is it something I can post on Webpack's Github?

**Comment by guybedford** [comment]
Yes it's a Webpack debugging issue, whether it is a Webpack bug specifically I don't know without investigating further though.

**Comment by kouts** [comment]
As an update, unfortunately, the ```script-loader``` solution does not work in IE11,
so the only solution, for now, is to manually copy systemjs files content into the bundle.

**Comment by guybedford** [comment]
The script loader is supposed to work in IE11.... are you getting incorrect registration timing issues here?

**Comment by kouts** [comment]
I wouldn't know how to check for that, I don't get any errors on the console though.
I have updated the [test repo](https://github.com/kouts/systemjs-webpack) which showcases the issue with the latest version of systemjs.

**Comment by guybedford** [comment]
So what exactly is the failure scenario here then?

**Comment by kouts** [comment]
I'm sorry maybe I didn't explain it clearly. When I say ```script-loader``` I'm referring to [Webpack script-loader](https://github.com/webpack-contrib/script-loader) which I used as a workaround to make the build work.

**Comment by guybedford** [comment]
The issue I see in the test repo is exactly the scenario from https://github.com/systemjs/systemjs/issues/1926#issuecomment-484590549. Is that the only problem here?

**Comment by kouts** [comment]
Yes, that is the problem, I just highlighted that what I used as a workaround does not work in IE (as a note to self and others watching this thread).

**Comment by guybedford** [comment]
I would suggest not using Webpack to build extras, and instead do a simple module concatenation - just concatenating the sources will always work out fine.
